aggregator/src/app/AggregationStrategy.ts

```diff
@@ -231,22 +236,36 @@ export default class AggregationStrategy {
       assert(after.success);
 
       const bundleResult = processBundleResults[i];
-
-      let success: boolean;
-
-      if (bundleResult.success) {
-        const [operationResults] = bundleResult.returnValue;
-
-        // We require that at least one operation succeeds, even though
-        // processBundle doesn't revert in this case.
-        success = operationResults.some((opSuccess) => opSuccess === true);
-      } else {
-        success = false;
+      const fee = after.returnValue[0].sub(before.returnValue[0]);
+      if (!bundleResult.success) {
+        return { success: false, fee, errorReason: null };
       }
 
-      const fee = after.returnValue[0].sub(before.returnValue[0]);
+      const [operationStatuses, results] = bundleResult.returnValue;
+
+      let errorReason: OperationResultError | null = null;
+      // We require that at least one operation succeeds, even though
+      // processBundle doesn't revert in this case.
+      const success = operationStatuses.some((opSuccess: boolean) => opSuccess === true);
+
+      // If operation is not successful, attempt to decode an error message
+      if (!success) {
+        const error = results.map((result: string[]) => {
+          try {
+            if (result[0]) {
+              return decodeError(result[0]);
+            }
+            return;
+          } catch (_err) {
+            return;
+          }
```

Comment:
Would be good to log out the error here. May even be worth adding and emitting as an `AppEvent`.

```suggestion
          } catch (err) {
            console.error(err);
            return;
          }
```